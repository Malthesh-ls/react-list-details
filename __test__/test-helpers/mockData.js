const mockRespUniversityData = [
  {
    "state-province": "Abu Dhabi",
    "domains": [
      "mbzuai.ac.ae"
    ],
    "web_pages": [
      "https://mbzuai.ac.ae/"
    ],
    "name": "Mohamed bin Zayed University of Artificial Intelligence (MBZUAI)",
    "alpha_two_code": "AE",
    "country": "United Arab Emirates"
  },
  {
    "domains": [
      "acd.ac.ae"
    ],
    "country": "United Arab Emirates",
    "state-province": null,
    "alpha_two_code": "AE",
    "web_pages": [
      "http://www.acd.ac.ae/"
    ],
    "name": "American College Of Dubai"
  },
];

const mockUniversityData = [
  {
    "state_province": "Abu Dhabi",
    "domains": [
      "mbzuai.ac.ae"
    ],
    "web_pages": [
      "https://mbzuai.ac.ae/"
    ],
    "name": "Mohamed bin Zayed University of Artificial Intelligence (MBZUAI)",
    "alpha_two_code": "AE",
    "country": "United Arab Emirates",
    "uid": 1
  },
  {
    "domains": [
      "acd.ac.ae"
    ],
    "country": "United Arab Emirates",
    "state_province": null,
    "alpha_two_code": "AE",
    "web_pages": [
      "http://www.acd.ac.ae/"
    ],
    "name": "American College Of Dubai",
    "uid": 3
  },
];

const mockColumns = [
  { "label": "Name", "accessor": "name", "sortable": true, "sortbyOrder": "asc" },
  { "label": "Alpha Code", "accessor": "alpha_two_code", "sortable": true, },
  { "label": "State Province", "accessor": "state_province", "sortable": true },
  { "label": "Country", "accessor": "country", "sortable": true },
  { "label": "Action", "accessor": "action", "sortable": false },
];

export {
  mockRespUniversityData,
  mockUniversityData,
  mockColumns,
}