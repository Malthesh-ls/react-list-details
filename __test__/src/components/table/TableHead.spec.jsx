import renderer from 'react-test-renderer';
import TableHead from '../../../../src/components/table/TableHead';
import { mockUniversityData, mockColumns } from '../../../test-helpers/mockData';

describe('TableHead Tests', () => {
  const props = {
    handleSorting: jest.fn(),
    columns: mockColumns,
  }
  it('should match snapshot', () => {
    const snapshot = renderer
      .create(<TableHead {...props} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});