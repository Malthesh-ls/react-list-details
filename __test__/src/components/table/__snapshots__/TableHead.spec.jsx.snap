// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead Tests should match snapshot 1`] = `
<thead>
  <tr>
    <th
      className="default"
      onClick={[Function]}
    >
      Name
    </th>
    <th
      className="default"
      onClick={[Function]}
    >
      Alpha Code
    </th>
    <th
      className="default"
      onClick={[Function]}
    >
      State Province
    </th>
    <th
      className="default"
      onClick={[Function]}
    >
      Country
    </th>
    <th
      className=""
      onClick={null}
    >
      Action
    </th>
  </tr>
</thead>
`;
